stdlib/array.mli

```diff
@@ -203,6 +203,11 @@ val for_all : ('a -> bool) -> 'a array -> bool
    [(p a1) && (p a2) && ... && (p an)].
    @since 4.03.0 *)
 
+val for_alli : (int -> 'a -> bool) -> 'a array -> bool
+(** Same as {!Array.for_all}, but the
+   function is applied with the index of the element as first argument,
+   and the element itself as second argument. *)
```

Comment:
(As mentioned in-thread, I would not mix stdlib extension and your feature in the same PR.)